setup.sh

```diff
@@ -42,26 +46,61 @@ if [[ -n $JAX_VERSION && ! ($MODE == "stable" || -z $MODE) ]]; then
 fi
 
 # Install JAX and JAXlib based on the specified mode
-if [[ "$MODE" == "stable" || ! -v MODE ]]; then
+if [[ "$MODE" == "pinned" ]]; then
+  if [[ "$DEVICE" != "gpu" ]]; then
+    echo "pinned mode is supported for GPU builds only."
+    exit 1
+  fi
+  echo "Installing pinned jax, jaxlib for NVIDIA gpu."
+  pip3 install "jax[cuda12_pip]" -f https://storage.googleapis.com/jax-releases/jax_cuda_releases.html -c constraints_gpu.txt
+  pip3 install "transformer-engine==1.5.0+297459b" \
```

Comment:
We don't need transoformer-engine for Maxdiffusion. This additional deps slows down the initial process a lot.